plugins/x86/x86_tools_prefix.ml

```diff
@@ -1,40 +1,14 @@
 open Core_kernel.Std
 open Bap.Std
-open X86_opcode
 open X86_tools_types
 
-module type W = module type of Word
 module Make (RR : RR) (FR : FR) (IV : IV) : PR = struct
-  type t = [
-    | `LOCK_PREFIX
-    | `REP_PREFIX
-    | `REPE_PREFIX
-    | `REPNE_PREFIX
-    | `EXCEPTION of int
-    | `NONE
-  ] [@@deriving sexp, compare]
-
-  let parse mem op =
-    let code = Memory.(mem ^! min_addr mem) |>
-               Word.to_int |> ok_exn in
-    match code, op with
-    | 0xF0, #btx_mi
-    | 0xF0, #btx_mr
-    | 0xF0, #cmpxchg_rm -> `LOCK_PREFIX
-    | 0xF0, _ -> `EXCEPTION (IV.to_int `UD)
-    | 0xF3, #ins
-    | 0xF3, #lods
-    | 0xF3, #movs
-    | 0xF3, #outs
-    | 0xF3, #stos -> `REP_PREFIX
-    | 0xF3, #cmps
-    | 0xF3, #scas -> `REPE_PREFIX
-    | 0xF2, #cmps
-    | 0xF2, #scas -> `REPNE_PREFIX
-    | _ -> `NONE
+  type t = X86_prefix.t [@@deriving sexp, compare]
 
   let lock bil = Bil.special "lock" :: bil
 
+  let exception_ n = [ Bil.cpuexn n ]
```

Comment:
do you really need this new definition? It doesn't introduce any abstractions and is even bigger  than its own body. I would call it Mr. Mackey definition :)